### PR TITLE
Mark `Flash` as deprecated

### PR DESCRIPTION
**Pull request by khiga8**
_Authors: Please fill out this form carefully and completely._

_Reviewers: By approving this Pull Request you are approving the code change, as well as its deployment and mitigation plans._
_Please read this description carefully. If you feel there is anything unclear or missing, please ask for updates._

### What are you trying to accomplish?
Formally deprecates `Flash` in favor of `Banner`.

[Related slack thread](https://github.slack.com/archives/CGJTTJ738/p1707774059572439?thread_ts=1707524004.597779&cid=CGJTTJ738)

### Integration
This may require the `DeprecatedComponentsCounter` to be updated.

#### List the issues that this change affects.
Fixes: https://github.com/primer/view_components/issues/2599

#### Risk Assessment
  

- [ ] **Low risk** the change is small, highly observable, and easily rolled back.
- [x] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### Anything you want to highlight for special attention from reviewers?

Do I need to do anything else?

### CODE REVIEW

File: .changeset/mighty-poems-know.md
Comment (khiga8) [inline .changeset/mighty-poems-know.md:2]: Is `patch` appropriate?

Comment (keithamus) [reply .changeset/mighty-poems-know.md:2]: I think we do minor now? I forget. /cc @camertron

Comment (camertron) [reply .changeset/mighty-poems-know.md:2]: Yeah, I think a minor bump is appropriate here.

**Comment by khiga8** [comment]
Updated to `minor`! Please feel free to merge this when ready since I don't have merge permissions. 🚀

**Review by camertron — APPROVED** [review]

**Pull request merged**
